cmd/helm/upgrade.go

```diff
@@ -226,6 +227,7 @@ func newUpgradeCmd(cfg *action.Configuration, out io.Writer) *cobra.Command {
 	f.BoolVar(&client.ReuseValues, "reuse-values", false, "when upgrading, reuse the last release's values and merge in any overrides from the command line via --set and -f. If '--reset-values' is specified, this is ignored")
 	f.BoolVar(&client.Wait, "wait", false, "if set, will wait until all Pods, PVCs, Services, and minimum number of Pods of a Deployment, StatefulSet, or ReplicaSet are in a ready state before marking the release as successful. It will wait for as long as --timeout")
 	f.BoolVar(&client.WaitForJobs, "wait-for-jobs", false, "if set and --wait enabled, will wait until all Jobs have been completed before marking the release as successful. It will wait for as long as --timeout")
+	f.IntVar(&client.WaitRetries, "wait-retries", 0, "if set and --wait enabled, will retry any failed check on resource state, except if HTTP status code < 500 is received, subject to the specified number of retries")
```

Comment:
I wonder if a flag here is needed? The client already specifies the wait timeout. And given a sensible retry policy, which only retries on retryable errors. Is it even needed to expose the fact that retries are happening to the user? Why not just continue to retry within the timeout? (or perhaps upto some sensible default)

Reply:
Removed flag, using a default instead. We discussed maybe setting 5 as the default number of retries in the weekly meeting. Given the polling happens every 2 seconds that would result in failure if the error condition lasts longer than 10 seconds. Im concerned this may be a bit short in some scenarios, so I have set to ma number of retires to 30 (i.e. one minute before failing). I can reduce again if this is considered to high